Project/Core/Logger.cs

```diff
@@ -39,6 +39,22 @@ private void CreateDirectoryIfNotExist()
         {
             System.IO.Directory.CreateDirectory(Configuration.AppConfig.LOG_DIRECTORY);
         }
+        private void DeleteLogFileIfBig()
+        {
+            // Deletes log file when it exists and is big enough.
+            if (System.IO.File.Exists(Configuration.AppConfig.LOG_FILE)
+                && new System.IO.FileInfo(Configuration.AppConfig.LOG_FILE).Length > Configuration.AppConfig.LOG_FILE_SIZE_LIMIT)

```

Comment:
Seems like you create FileInfo every time on writing to log. A lot of unused memory, can you save it as a field, and replace all **File** method with **FileInfo**

Reply:
Okay